Formula/oil.rb

```diff
@@ -1,11 +1,16 @@
 class Oil < Formula
   desc "Bash-compatible Unix shell with more consistent syntax and semantics"
   homepage "https://www.oilshell.org/"
-  url "https://www.oilshell.org/download/oil-0.8.3.tar.gz"
-  sha256 "463efc219af8e77eb0b0d183feaa357f4d8e7c3813382055ef262c91a0c44092"
+  url "https://www.oilshell.org/download/oil-0.8.5.tar.gz"
+  sha256 "4286eebb190f020e5c2472a48b0aa16a5abecfbf60068d1d9ad57d694e3ffc0a"
   license "Apache-2.0"
   head "https://github.com/oilshell/oil.git"
 
+  livecheck do
+    url "https://www.oilshell.org/releases.html"
+    regex(/class="version-number">v?(\d+(?:\.\d+)+)</i)
```

Comment:
```suggestion
    regex(/href=.*?oil[._-]v?(\d+(?:\.\d+)+)\.t/i)
```

Is there a reason to specifically match the version numbers using the `.version-number` elements? To me, it seems like we should simply identify versions using the tarball URLs on the page like we normally do.

Reply:
There really isn't one except that you are dealing with lazy dev whose idea was to model it after SQLite formula and call it a day. 😁 
Thanks for pointing me in the right direction, I'm still learning those livecheck idioms.

Reply:
Ahah, no worries. I just wanted to make sure I wasn't overlooking something important.

`sqlite` is unfortunately a bit of an outlier, so it's just bad luck that you didn't happen to pick one of the hundreds of formulae that use this common regex pattern (i.e., `/href=.*?example[._-]v?(\d+(?:\.\d+)+)\.t/i`, where `example` is usually the software name).

Reply:
Speaking of, IIRC someone mentioned defining a standard set of regexes that could be referenced in formulas? Is that just my wishful thinking or I really read that somewhere?

Reply:
I have plans to improve the ergonomics of writing `livecheck` blocks in the future (once I work through some items on my lengthy livecheck to-do list). Building standard snippets like these into livecheck (in some fashion) is the main part of that work.

We currently have a lot of similar regexes in formulae that follow patterns like these but this is by design, as it's a step towards built-in snippets. A few months ago, we did a lot of work on standardizing regexes (where possible) and bringing existing `livecheck` blocks closer to our current standards, so it's all part of the forward march of progress.

Once I finish the livecheck documentation PR, I'll be back working on improving livecheck's internals again and there's some good stuff in the pipeline.

Reply:
> We currently have a lot of similar regexes in formulae that follow patterns like these but this is by design, as it's a step towards built-in snippets. A few months ago, we did a lot of work on standardizing regexes (where possible) and bringing existing `livecheck` blocks closer to our current standards, so it's all part of the forward march of progress.

Sounds like a well-thought plan! 👍 

> Once I finish the livecheck documentation PR, I'll be back working on improving livecheck's internals again and there's some good stuff in the pipeline.

I'll gladly join the hacking effort and thank you for all livecheck goodies! 👍